cmd/crank/beta/xpkg/init.go

```diff
@@ -116,14 +138,127 @@ func (c *initCmd) Run(k *kong.Context, logger logging.Logger) error {
 		return errors.Wrapf(err, "failed to get repository's HEAD from %q", repoURL)
 	}
 
-	// TODO(phisco): replace placeholders for the name all around the
-	// 	repository? Maybe we can just agree on some markdown text in the
-	// 	repos to print to let the user know what to do next?
-
 	_, err = fmt.Fprintf(k.Stdout, "Initialized package %q in directory %q from %s (%s)\n", c.Name, c.Directory, getPrettyURL(logger, repoURL, ref), ref.Name().Short())
+	if err != nil {
+		return errors.Wrap(err, "failed to write to stdout")
+	}
+
+	err = handleNotes(k.Stdout, logger, c.Directory)
+	if err != nil {
+		return errors.Wrap(err, "failed to handle NOTES.txt")
+	}
+
+	err = c.handleInitScript(k, logger, c.Directory, c.Name)
+	if err != nil {
+		return errors.Wrap(err, "failed to handle initialization script")
+	}
+
 	return err
 }
 
+// handleNotes prints the NOTES.txt file in the template
+// repository, if it exists.
+func handleNotes(w io.Writer, logger logging.Logger, dir string) error {
+	notesFile := dir + "/" + notes
+	f, err := os.Stat(notesFile)
+	switch {
+	case os.IsNotExist(err):
+		// no NOTES.txt file, skip
+		logger.Debug("No NOTES.txt found, skipping")
+		return nil
+	case err != nil:
+		return errors.Wrapf(err, "failed to stat notes file %s", notesFile)
+	case f.IsDir():
+		return errors.Errorf("%s is not a file", notesFile)
+	}
+
+	return errors.Wrapf(printFile(w, notesFile), "failed to print file %s", notesFile)
+}
+
+// handleInitScript runs the init.sh script in the template repository, if it
+// exists.
+func (c *initCmd) handleInitScript(k *kong.Context, logger logging.Logger, dir, name string) error {
+	scriptFile := dir + "/" + initScript
+	f, err := os.Stat(scriptFile)
+	switch {
+	case os.IsNotExist(err):
+		// no init.sh file, skip
+		logger.Debug("No init.sh found, skipping")
+		return nil
+	case err != nil:
+		return errors.Wrapf(err, "failed to stat init.sh file %s", scriptFile)
+	case f.IsDir():
+		return errors.Errorf("%s is not a file", scriptFile)
+	}
+
+	if c.RunInitScript {
+		return errors.Wrapf(runScript(k, scriptFile, name, dir), "failed to run init script %s", scriptFile)
+	}
+
+	// ask the user if they want to display the script
+	display, err := promptYesNo(k, "\nFound init.sh script. Do you want to display it?")
+	if err != nil {
+		return errors.Wrap(err, "failed to prompt user")
+	}
+	if display {
+		err = printFile(k.Stdout, scriptFile)
+		if err != nil {
+			return errors.Wrapf(err, "failed to print file %s", scriptFile)
+		}
+	}
+
+	// ask the user if they want to run the script
+	run, err := promptYesNo(k, "\nDo you want to run the script?")
+	if err != nil {
+		return errors.Wrap(err, "failed to prompt user")
+	}
+	if run {
+		return errors.Wrapf(runScript(k, scriptFile, name, dir), "failed to run init script %s", scriptFile)
+	}
+	return nil
```

Comment:
Nit: what about a single prompt like `Found init.sh script. Do you want to run it? [y]es/[n]o/[v]iew`

Reply:
if they select view, what would happen after we display it? would we prompt them again?

Reply:
I was thinking to do that, if they select view, just display and prompt again.

Reply:
Looks like this now:
```
 ~/go/crossplane/_output/bin/darwin_arm64/crank beta xpkg init func https://github.com/lsviben/function-template-go -d func16
Initialized package "func" in directory "/tmp/t/func16" from https://github.com/lsviben/function-template-go/tree/a5e76778657cd80bfe8100e77608f0577f75e291 (main)

To get started:

1. Replace `function-template-go` with {{ .Name }} in `go.mod`,
   `package/crossplane.yaml`, and any Go imports
2. Update `input/v1beta1/` to reflect your desired input (and run `go generate`)
3. Add your logic to `RunFunction` in `fn.go`
4. Add tests for your logic in `fn_test.go`
5. Update `README.md`, to be about your function!

Found init.sh script!
Do you want to run it? [y]es/[n]o/[v]iew: v

#!/bin/sh

echo hello "$1"
Do you want to run it? [y]es/[n]o/[v]iew: y
hello func
```

The same prompt is repeated after choosing "view", in theory you could view it forever if somebody wanted, but I dont think its an issue.